graaljs/graaljs-gradle-webpack-guide/README.md

```diff
@@ -0,0 +1,187 @@
+
+    // Get QRCode object and generate QR code as a string
+    QRCode qrCode = exports.getMember("QRCode").as(QRCode.class);
+    Promise resultPromise = qrCode.toString(input);
+
+    resultPromise.then((result) -> {
+        System.out.println("Successfully generated QR code for \"" + input + "\".");
+        System.out.println(result.asString());
+    });
+}
+```
+
+### `QRCode.java`
+```java
+public interface QRCode {
+    Promise toString(String data);
+    Promise toDataURL(String data, Object options);
+}
+```
+
+### `Promise.java`
+```java
+public interface Promise {
+    Promise then(ValueConsumer onResolve);
+    Promise then(ValueConsumer onResolve, ValueConsumer onReject);
+}
+```
+
+## JavaScript Code Overview
+
+### `main.mjs`
+```js
+import 'fast-text-encoding';
+export * as QRCode from 'qrcode';
+```
+
+### Webpack Configuration
+
+#### `webpack.config.js`
+```js
+const path = require('path');
+const { EnvironmentPlugin } = require('webpack');
+
+const isProduction = process.env.NODE_ENV == 'production';
+
+const config = {
+    entry: './main.mjs',
+    output: {
+        path: process.env.BUILD_DIR
+            ? path.resolve(process.env.BUILD_DIR)
+            : path.resolve(__dirname, '../../build/classes/java/main/bundle'),
+        filename: 'bundle.mjs',
+        module: true,
+        library: {
+            type: 'module',
+        },
+        globalObject: 'globalThis'
+    },
+    experiments: {
+        outputModule: true
+    },
+    mode: isProduction ? 'production' : 'development',
+};
+
+module.exports = () => config;
+```
+
+## Gradle Configuration
+
+#### `build.gradle`
+This Gradle configuration sets up the Java build environment, including the necessary dependencies for GraalVM and Node.js.
+
+```gradle
+plugins {
+    id 'java'
+    id 'application'
+    id 'com.github.node-gradle.node' version '7.0.1'
+}
+
+group = 'org.example'
+version = '1.0-SNAPSHOT'
+
+repositories {
+    mavenCentral()
+}
+
+application {
+    mainClass = 'org.example.App'
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(21))
+    }
+}
+
+dependencies {
+    implementation 'org.graalvm.polyglot:polyglot:24.2.0'
+    implementation 'org.graalvm.polyglot:js:24.2.0'
+    
+    testImplementation platform('org.junit:junit-bom:5.11.0')
+    testImplementation 'org.junit.jupiter:junit-jupiter-api'
+    testImplementation 'org.junit.jupiter:junit-jupiter-params'
+}
+
+test {
+    useJUnitPlatform()
+}
+
+node {
+    version = '22.14.0'
+    npmVersion = '10.9.2'
+    download = true
+    workDir = file("${project.buildDir}/node")
+    npmWorkDir = file("${project.buildDir}/npm")
+    nodeProjectDir = file('src/main/js')
+}
+
+tasks.register('webpackBuild', NpmTask) {
+    dependsOn tasks.npmInstall
+    workingDir = file('src/main/js')
+    args = ['run', 'build']
+    environment = ['BUILD_DIR': "${buildDir}/classes/java/main/bundle"]
+}
+
+processResources.dependsOn tasks.webpackBuild
```

Comment:
This and the plugin registration (`id 'com.github.node-gradle.node' version '7.0.1'`) is needed to set up Node and webpack, similar to https://github.com/graalvm/graal-languages-demos/blob/0951909cb64a04262eaa6a27bc5800846e785019/graaljs/graaljs-maven-webpack-guide/README.md#L73-L113.